src/test/regress/specs/isolation_replace_wait_function.spec

```diff
@@ -0,0 +1,53 @@
+# check that replace_isolation_tester_func correctly replaces the functions isolation
+# tester uses while searching for locks. If those functions aren't correctly replaced
+# this test will timeout, since isolation tester will never notice that s2 is blocked
+# by s1 on a lock it's taken out on one of the workers
+
+setup
+{
+  SELECT citus.replace_isolation_tester_func();
+
+  CREATE TABLE test_locking (a int unique);
+  SELECT create_distributed_table('test_locking', 'a');
+
+  -- above we replaced the function isolation tester uses, however isolation tester has
+  -- already created a prepared statement with the old function. By changing the
+  -- search_path we force a re-parse which will pick up the new definition
+  SET search_path = 'citus';
```

Comment:
Is there a way in which we could achieve this as part of `replace_isolation_tester_func`?